### DISCUSSION TITLE
New Year, New Check-In: Here’s The Latest API and Webhooks Community Check-in 🎉

### DISCUSSION THREAD

**Discussion by ghostinhershell** [community/community#88438] [Apps, API and Webhooks]
Hey hey! 👋 Reporting in for our first API and Webhooks Community Check-In for 2024 🎆

### Big Kudos to these Superstar Community Contributors 👏
- Shout out to @DozenLimeStone for helping @xts-bit figure out [how to upload a folder that contained all of their code files to their Github repo using Github's REST API](https://github.com/orgs/community/discussions/83252). Also, nice user handle!
- @rfossicav was able to find the answer to his question regarding [a graphQL RuleSource node / sourceID global id error](https://github.com/orgs/community/discussions/77974) by chiming into another, similar discussion, and getting help there. Big kudos to @jschnapper for hopping in and lending a helping hand!

Have you been helped by a community superstar that you’d like to see celebrated? Tag them to nominate them below!

For community members who post questions, [marking responses that solves your question as the answer](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer) goes a long way. When you mark a comment in the Discussion as an answer, GitHub will highlight the comment and replies to the comment to help other community members quickly find the answer as well.

![The image is a screenshot showing where users can click to mark a response as the answer](https://github.com/community/community/assets/14021469/01d2f14c-6a27-4654-8522-0f72463bf0d5)

Remember to thank these and other community members that take the time to comment and help troubleshoot these problems. An acknowledgement of a job well done can have an incredible effect and make someone’s day. ☀️ We’re here to learn, discover and connect so try and mark a comment as an answer if suitable, give a thank you and select a reaction as a way to show your appreciation! 

## ⬆️Product Updates: ICYMI!
- **[Upcoming changes to repository insights](https://github.blog/changelog/2023-11-29-upcoming-changes-to-repository-insights/)** - Beginning January 8th, 2024, we will be making changes to the repository insights UI and API on GitHub for repositories with over 10,000 commits.
- **[Creating temporary private forks via the REST API](https://docs.github.com/en/rest/security-advisories/repository-advisories?apiVersion=2022-11-28)** - You can now use the REST API to create a temporary private fork within a draft security advisory or private vulnerability report.
- **[Webhook delivery logs will only be retained for 3 days](https://github.blog/changelog/2023-10-17-webhook-delivery-logs-will-only-be-retained-for-3-days/)** - For up to 3 days after a webhook delivery is posted, you can view details about that delivery and request a redelivery.
- **[Repository security advisories REST API](https://github.blog/changelog/2023-03-30-repository-security-advisories-rest-api/)** - You can now programmatically view and act on repository advisories via a new REST API. New endpoints to create, view, list, and update advisories are available to all. Additionally, new webhooks have been introduced that will alert maintainers when advisories are published or when a private vulnerability report is submitted.

## 🗞️What’s New?
- **[Our move to generated SDKs](https://github.blog/2024-01-03-our-move-to-generated-sdks/)** - Taking a leap: Moving away from our traditional Octokit SDKs and introducing our generated Go and .NET SDKs.
- **[New Markdown extension](https://github.blog/changelog/2023-12-14-new-markdown-extension-alerts-provide-distinctive-styling-for-significant-content/)** -  Alerts provide distinctive styling for significant content.
- **Company Update: Diversity, inclusion, and belonging at GitHub in 2023** - We expanded the diversity of our employee base, launched several new employee engagement programs, made investments to improve accessibility, and more. Read more about it on [The GitHub Blog](​​https://github.blog/2023-11-01-diversity-inclusion-and-belonging-at-github-in-2023/).

— 
A very big thank you to everyone in the community who actively contributes thoughts and answers to the questions that come into the API and Webhooks category. Looking forward to seeing more Community contributions to come in 2024!

**Comment by ghostinhershell** [comment]
Hey there 👋🏾

Missed out on our last community check-in? You can check that out here: [API & Webhooks: December Community Check-In ⛄](https://github.com/orgs/community/discussions/80939).

**Comment by DozenLimeStone** [comment]
🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ 🏆​​ vv
